homeassistant/components/flo/__init__.py

```diff
@@ -0,0 +1,80 @@
+"""The flo integration."""
+import asyncio
+import logging
+
+from aioflo import async_get_api
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+from .device import FloDevice
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({})}, extra=vol.ALLOW_EXTRA)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["sensor"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the flo component."""
+    hass.data[DOMAIN] = {}
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up flo from a config entry."""
+    hass.data[DOMAIN][entry.entry_id] = {}
+    session = async_get_clientsession(hass)
+    hass.data[DOMAIN][entry.entry_id]["client"] = client = await async_get_api(
```

Comment:
Possible to catch bad username/password here? Can we raise ConfigEntryNotReady if we can't connect?

Reply:
updated